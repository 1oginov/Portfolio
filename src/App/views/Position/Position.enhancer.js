/* @flow */

import { connect } from 'react-redux';
import { compose, withHandlers, type HOC } from 'recompose';

import { deletePosition } from 'Portfolio/actions';
import withPositionById from 'Portfolio/enhancers/withPositionById';

import withNavigationHandlers from '../../enhancers/withNavigationHandlers';
import withRouteParams from '../../enhancers/withRouteParams';
import * as R from '../../routes';

const mapDispatchToProps = { deletePosition };

const enhancer: HOC<*, *> = compose(
  withNavigationHandlers({
    handleHomeClick: R.HOME,
  }),
  withRouteParams,
  withPositionById(({ routeParams }) => routeParams.position),
  connect(null, mapDispatchToProps),
  withHandlers({

    handleDeleteClick: ({
      deletePosition, handleHomeClick, position,
    }) => () => {
      deletePosition(position.id)
        .then(() => handleHomeClick());
    },

  }),
);

export default enhancer;
