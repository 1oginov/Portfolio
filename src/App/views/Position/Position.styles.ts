import { withStyles } from '@material-ui/core';

import { viewRoot } from 'Styles/mixins';

export default withStyles(theme => ({
  bar: {
    alignItems: 'center',
    borderBottom: `1px solid ${theme.palette.divider}`,
    display: 'flex',
    paddingBottom: theme.spacing(2),
    paddingLeft: theme.spacing(4),
    paddingRight: theme.spacing(2.5),
    paddingTop: theme.spacing(2),
  },
  companyName: {
    flexGrow: 1,
  },
  list: {
    width: '100%',
  },
  logo: {
    marginRight: theme.spacing(1.5),
  },
  root: {
    ...viewRoot(theme),
    padding: theme.spacing(2),
  },
}));
