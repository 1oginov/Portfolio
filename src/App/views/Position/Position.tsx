import {
  IconButton, List, ListItem, ListItemSecondaryAction, ListItemText, ListSubheader, Typography,
} from '@material-ui/core';
import { MonetizationOnOutlined, DeleteForeverOutlined, DeleteOutlined } from '@material-ui/icons';
import * as React from 'react';

import PositionDate from 'Portfolio/components/PositionDate';
import Money from 'Shared/components/Money';
import Percent from 'Shared/components/Percent';
import Progress from 'Shared/components/Progress';
import StockLogo from 'Stocks/components/StockLogo';

import { StockPosition } from '../../lib';

export interface Props {
  classes: { [key: string]: string };
  handleCloseClick: () => void;
  handleDeleteClick: () => void;
  handleWantToDelete: () => void;
  positionLoading: boolean;
  stockPosition: StockPosition | null;
  wantToDelete: boolean;
}

const Position: React.FunctionComponent<Props> = ({
  classes, handleCloseClick, handleDeleteClick, handleWantToDelete, positionLoading, stockPosition, wantToDelete,
}: Props) => (
  <React.Fragment>

    {!positionLoading && stockPosition && (
      <div className={classes.bar}>

        <StockLogo className={classes.logo} size={48} symbol={stockPosition.symbol} />

        {stockPosition.quote && (
          <Typography className={classes.companyName}>{stockPosition.quote.companyName}</Typography>
        )}

        {stockPosition.closeDate === null && (
          <IconButton onClick={handleCloseClick}>
            <MonetizationOnOutlined />
          </IconButton>
        )}

        {wantToDelete
          ? <IconButton onClick={handleDeleteClick}><DeleteForeverOutlined /></IconButton>
          : <IconButton onClick={handleWantToDelete}><DeleteOutlined /></IconButton>}

      </div>
    )}

    <div className={classes.root}>

      {positionLoading && <Progress />}

      {!positionLoading && stockPosition && (
        <List className={classes.list} dense>

          <ListItem>
            <ListItemText>Symbol</ListItemText>
            <ListItemSecondaryAction>{stockPosition.symbol}</ListItemSecondaryAction>
          </ListItem>
          <ListItem>
            <ListItemText>Amount</ListItemText>
            <ListItemSecondaryAction>{stockPosition.amount}</ListItemSecondaryAction>
          </ListItem>

          <ListSubheader disableSticky>Open</ListSubheader>
          <ListItem>
            <ListItemText>Date</ListItemText>
            <ListItemSecondaryAction>
              <PositionDate date={stockPosition.openDate} />
            </ListItemSecondaryAction>
          </ListItem>
          <ListItem>
            <ListItemText>Price</ListItemText>
            <ListItemSecondaryAction>
              <Money value={stockPosition.openPrice} />
            </ListItemSecondaryAction>
          </ListItem>
          <ListItem>
            <ListItemText>Commission</ListItemText>
            <ListItemSecondaryAction>
              <Money value={stockPosition.openCommission} />
            </ListItemSecondaryAction>
          </ListItem>
          <ListItem>
            <ListItemText>Sum</ListItemText>
            <ListItemSecondaryAction>
              <Money value={stockPosition.openSum} />
            </ListItemSecondaryAction>
          </ListItem>

          {stockPosition.closeDate !== null && stockPosition.closeCommission !== null
          && stockPosition.closePrice !== null && stockPosition.closeSum !== null && stockPosition.closePL !== null
          && stockPosition.closePLPercent !== null && stockPosition.closePLAnnualPercent !== null && (
            <React.Fragment>

              <ListSubheader disableSticky>Close</ListSubheader>
              <ListItem>
                <ListItemText>Date</ListItemText>
                <ListItemSecondaryAction>
                  <PositionDate date={stockPosition.closeDate} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>Price</ListItemText>
                <ListItemSecondaryAction>
                  <Money value={stockPosition.closePrice} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>Commission</ListItemText>
                <ListItemSecondaryAction>
                  <Money value={stockPosition.closeCommission} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>Sum</ListItemText>
                <ListItemSecondaryAction>
                  <Money value={stockPosition.closeSum} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>PL</ListItemText>
                <ListItemSecondaryAction>
                  <Money pl value={stockPosition.closePL} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>PL%</ListItemText>
                <ListItemSecondaryAction>
                  <Percent pl value={stockPosition.closePLPercent} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>Annual PL%</ListItemText>
                <ListItemSecondaryAction>
                  <Percent pl value={stockPosition.closePLAnnualPercent} />
                </ListItemSecondaryAction>
              </ListItem>

            </React.Fragment>
          )}

          {stockPosition.dailyPL !== null && stockPosition.dailyPLPercent !== null && (
            <React.Fragment>

              <ListSubheader disableSticky>Daily</ListSubheader>
              <ListItem>
                <ListItemText>PL</ListItemText>
                <ListItemSecondaryAction>
                  <Money pl value={stockPosition.dailyPL} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>PL%</ListItemText>
                <ListItemSecondaryAction>
                  <Percent pl value={stockPosition.dailyPLPercent} />
                </ListItemSecondaryAction>
              </ListItem>

            </React.Fragment>
          )}

          {stockPosition.marketPrice !== null && stockPosition.marketSum !== null
          && stockPosition.marketPL !== null && stockPosition.marketPLPercent !== null
          && stockPosition.marketPLAnnualPercent !== null && (
            <React.Fragment>

              <ListSubheader disableSticky>Market</ListSubheader>
              <ListItem>
                <ListItemText>Price</ListItemText>
                <ListItemSecondaryAction>
                  <Money value={stockPosition.marketPrice} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>Sum</ListItemText>
                <ListItemSecondaryAction>
                  <Money value={stockPosition.marketSum} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>PL</ListItemText>
                <ListItemSecondaryAction>
                  <Money pl value={stockPosition.marketPL} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>PL%</ListItemText>
                <ListItemSecondaryAction>
                  <Percent pl value={stockPosition.marketPLPercent} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>Annual PL%</ListItemText>
                <ListItemSecondaryAction>
                  <Percent pl value={stockPosition.marketPLAnnualPercent} />
                </ListItemSecondaryAction>
              </ListItem>

            </React.Fragment>
          )}

        </List>
      )}

    </div>

  </React.Fragment>
);

export default Position;
