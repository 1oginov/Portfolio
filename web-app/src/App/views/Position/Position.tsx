import {
  IconButton, List, ListItem, ListItemSecondaryAction, ListItemText, ListSubheader, Typography,
} from '@material-ui/core';
import {
  MonetizationOnOutlined, DeleteForeverOutlined, DeleteOutlined, EditOutlined,
} from '@material-ui/icons';
import * as React from 'react';

import { VibrantPalette } from 'Layers/Business/Services/FirebaseFunctionsService/VibrantPalette';
import { getVibrantColor } from 'Layers/Business/Services/FirebaseService/FirebaseService';
import { Money } from 'Layers/Presentation/Components/Money';
import { Percent } from 'Layers/Presentation/Components/Percent';
import { PositionDate } from 'Layers/Presentation/Components/PositionDate';
import { Progress } from 'Layers/Presentation/Components/Progress';
import { StockLogo } from 'Layers/Presentation/Components/StockLogo';

import { StockPosition } from '../../lib';

export interface Props {
  classes: { [key: string]: string };
  handleCloseClick: () => void;
  handleDeleteClick: () => void;
  handleUpdateClick: () => void;
  handleWantToDelete: () => void;
  positionLoading: boolean;
  stockPosition: StockPosition | null;
  vibrantPalette: VibrantPalette | null;
  wantToDelete: boolean;
}

const Position: React.FunctionComponent<Props> = ({
  classes, handleCloseClick, handleDeleteClick, handleUpdateClick, handleWantToDelete, positionLoading, stockPosition,
  vibrantPalette, wantToDelete,
}: Props) => (
  <>

    {!positionLoading && stockPosition && (
      <div className={classes.bar} style={{ backgroundColor: getVibrantColor(vibrantPalette, 'darkMuted') }}>

        <StockLogo className={classes.logo} size={48} symbol={stockPosition.symbol} />

        <Typography className={classes.companyName}>{stockPosition.companyName}</Typography>

        {stockPosition.closeDate === null && (
          <IconButton onClick={handleCloseClick}>
            <MonetizationOnOutlined />
          </IconButton>
        )}

        <IconButton onClick={handleUpdateClick}>
          <EditOutlined />
        </IconButton>

        {wantToDelete
          ? <IconButton onClick={handleDeleteClick}><DeleteForeverOutlined /></IconButton>
          : <IconButton onClick={handleWantToDelete}><DeleteOutlined /></IconButton>}

      </div>
    )}

    <div className={classes.root}>

      {positionLoading && <Progress />}

      {!positionLoading && stockPosition && (
        <List className={classes.list} disablePadding dense>

          <ListItem>
            <ListItemText>Symbol</ListItemText>
            <ListItemSecondaryAction>{stockPosition.symbol}</ListItemSecondaryAction>
          </ListItem>
          <ListItem>
            <ListItemText>Amount</ListItemText>
            <ListItemSecondaryAction>{stockPosition.amount}</ListItemSecondaryAction>
          </ListItem>

          <ListSubheader disableSticky>Open</ListSubheader>
          <ListItem>
            <ListItemText>Date</ListItemText>
            <ListItemSecondaryAction>
              <PositionDate date={stockPosition.openDate} />
            </ListItemSecondaryAction>
          </ListItem>
          <ListItem>
            <ListItemText>Price</ListItemText>
            <ListItemSecondaryAction>
              <Money value={stockPosition.openPrice} />
            </ListItemSecondaryAction>
          </ListItem>
          <ListItem>
            <ListItemText>Commission</ListItemText>
            <ListItemSecondaryAction>
              <Money value={stockPosition.openCommission} />
            </ListItemSecondaryAction>
          </ListItem>
          <ListItem>
            <ListItemText>Sum</ListItemText>
            <ListItemSecondaryAction>
              <Money value={stockPosition.openSum} />
            </ListItemSecondaryAction>
          </ListItem>

          {stockPosition.closeDate !== null && stockPosition.closeCommission !== null
          && stockPosition.closePrice !== null && stockPosition.closeSum !== null && stockPosition.closePL !== null
          && stockPosition.closePLPercent !== null && stockPosition.closePLAnnualPercent !== null && (
            <>

              <ListSubheader disableSticky>Close</ListSubheader>
              <ListItem>
                <ListItemText>Date</ListItemText>
                <ListItemSecondaryAction>
                  <PositionDate date={stockPosition.closeDate} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>Price</ListItemText>
                <ListItemSecondaryAction>
                  <Money value={stockPosition.closePrice} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>Commission</ListItemText>
                <ListItemSecondaryAction>
                  <Money value={stockPosition.closeCommission} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>Sum</ListItemText>
                <ListItemSecondaryAction>
                  <Money value={stockPosition.closeSum} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>PL</ListItemText>
                <ListItemSecondaryAction>
                  <Money pl value={stockPosition.closePL} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>PL%</ListItemText>
                <ListItemSecondaryAction>
                  <Percent pl value={stockPosition.closePLPercent} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>Annual PL%</ListItemText>
                <ListItemSecondaryAction>
                  <Percent pl value={stockPosition.closePLAnnualPercent} />
                </ListItemSecondaryAction>
              </ListItem>

            </>
          )}

          {stockPosition.dailyPL !== null && stockPosition.dailyPLPercent !== null && (
            <>

              <ListSubheader disableSticky>Daily</ListSubheader>
              <ListItem>
                <ListItemText>PL</ListItemText>
                <ListItemSecondaryAction>
                  <Money pl value={stockPosition.dailyPL} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>PL%</ListItemText>
                <ListItemSecondaryAction>
                  <Percent pl value={stockPosition.dailyPLPercent} />
                </ListItemSecondaryAction>
              </ListItem>

            </>
          )}

          {stockPosition.marketPrice !== null && stockPosition.marketSum !== null
          && stockPosition.marketPL !== null && stockPosition.marketPLPercent !== null
          && stockPosition.marketPLAnnualPercent !== null && (
            <>

              <ListSubheader disableSticky>Market</ListSubheader>
              <ListItem>
                <ListItemText>Price</ListItemText>
                <ListItemSecondaryAction>
                  <Money value={stockPosition.marketPrice} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>Sum</ListItemText>
                <ListItemSecondaryAction>
                  <Money value={stockPosition.marketSum} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>PL</ListItemText>
                <ListItemSecondaryAction>
                  <Money pl value={stockPosition.marketPL} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>PL%</ListItemText>
                <ListItemSecondaryAction>
                  <Percent pl value={stockPosition.marketPLPercent} />
                </ListItemSecondaryAction>
              </ListItem>
              <ListItem>
                <ListItemText>Annual PL%</ListItemText>
                <ListItemSecondaryAction>
                  <Percent pl value={stockPosition.marketPLAnnualPercent} />
                </ListItemSecondaryAction>
              </ListItem>

            </>
          )}

        </List>
      )}

    </div>

  </>
);

export default Position;
