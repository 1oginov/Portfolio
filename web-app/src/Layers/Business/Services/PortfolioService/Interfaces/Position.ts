export interface Position {
  id: string;
  symbol: string;
  amount: number;

  openPrice: number;
  openCommission: number;
  openDate: string;

  closePrice: number | null;
  closeCommission: number | null;
  closeDate: string | null;
}
