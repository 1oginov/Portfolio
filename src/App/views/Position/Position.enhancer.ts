import { connect } from 'react-redux';
import { compose, withHandlers } from 'recompose';

import { deletePosition as deletePositionAction, DeletePositionAction } from 'Portfolio/actions';
import withPositionById from 'Portfolio/enhancers/withPositionById';
import { Position } from 'Portfolio/lib';

import withNavigationHandlers from '../../enhancers/withNavigationHandlers';
import withRouteParams from '../../enhancers/withRouteParams';
import { Props } from './Position';
import * as R from '../../routes';
import { RouteParamsState } from '../../State';

interface WithHandlersProps {
  deletePosition: DeletePositionAction;
  handleClosePositionClick: (positionId: string) => void;
  handleHomeClick: () => void;
  position: Position | null;
  positionLoading: boolean;
}

const mapDispatchToProps = { deletePosition: deletePositionAction };

export default compose<Props & WithHandlersProps, {}>(
  withNavigationHandlers({
    handleHomeClick: R.HOME,
  }),
  withRouteParams,
  withPositionById<{ routeParams: RouteParamsState }>(({ routeParams }) => routeParams.position),
  withNavigationHandlers({
    handleCloseClick: ({ position }) => ({
      params: { position: position.id },
      route: R.CLOSE_POSITION,
    }),
  }),
  connect(null, mapDispatchToProps),
  withHandlers<WithHandlersProps, {}>({

    handleDeleteClick: ({
      deletePosition, handleHomeClick, position,
    }) => () => {
      if (position) {
        deletePosition(position.id)
          .then(() => handleHomeClick());
      }
    },

  }),
);
