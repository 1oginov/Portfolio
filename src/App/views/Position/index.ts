import Position from './Position';
import enhance from './Position.enhancer';

export default enhance(Position);
