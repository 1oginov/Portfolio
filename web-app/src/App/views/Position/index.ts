import Position from './Position';
import enhance from './Position.enhancer';
import style from './Position.styles';

export default style(enhance(Position));
