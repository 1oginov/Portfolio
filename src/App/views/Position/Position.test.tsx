import { shallow } from 'enzyme';
import * as React from 'react';

import Position from './Position';

it('matches snapshot', () => {
  const wrapper = shallow(
    <Position
      handleDeleteClick={() => undefined}
      handleHomeClick={() => undefined}
      position={{
        amount: 1,
        date: '2019-07-01',
        id: 'id',
        price: 100,
        symbol: 'symbol',
      }}
      positionLoading={false}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
