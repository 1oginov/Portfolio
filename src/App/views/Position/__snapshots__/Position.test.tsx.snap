// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <div
    className="bar"
  >
    <WithStyles(Connect(lifecycle(mapProps(StockLogo))))
      className="logo"
      size={48}
      symbol="AAPL"
    />
    <WithStyles(ForwardRef(Typography))
      className="companyName"
    >
      Apple, Inc.
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(IconButton))
      onClick={[Function]}
    >
      <MonetizationOnOutlinedIcon />
    </WithStyles(ForwardRef(IconButton))>
    <WithStyles(ForwardRef(IconButton))
      onClick={[Function]}
    >
      <EditOutlinedIcon />
    </WithStyles(ForwardRef(IconButton))>
    <WithStyles(ForwardRef(IconButton))
      onClick={[Function]}
    >
      <DeleteOutlinedIcon />
    </WithStyles(ForwardRef(IconButton))>
  </div>
  <div
    className="root"
  >
    <WithStyles(ForwardRef(List))
      className="list"
      dense={true}
      disablePadding={true}
    >
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemText))>
          Symbol
        </WithStyles(ForwardRef(ListItemText))>
        <WithStyles(ForwardRef(ListItemSecondaryAction))>
          AAPL
        </WithStyles(ForwardRef(ListItemSecondaryAction))>
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemText))>
          Amount
        </WithStyles(ForwardRef(ListItemText))>
        <WithStyles(ForwardRef(ListItemSecondaryAction))>
          1
        </WithStyles(ForwardRef(ListItemSecondaryAction))>
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListSubheader))
        disableSticky={true}
      >
        Open
      </WithStyles(ForwardRef(ListSubheader))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemText))>
          Date
        </WithStyles(ForwardRef(ListItemText))>
        <WithStyles(ForwardRef(ListItemSecondaryAction))>
          <PositionDate
            date="2019-07-01"
          />
        </WithStyles(ForwardRef(ListItemSecondaryAction))>
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemText))>
          Price
        </WithStyles(ForwardRef(ListItemText))>
        <WithStyles(ForwardRef(ListItemSecondaryAction))>
          <WithStyles(Money)
            value={100}
          />
        </WithStyles(ForwardRef(ListItemSecondaryAction))>
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemText))>
          Commission
        </WithStyles(ForwardRef(ListItemText))>
        <WithStyles(ForwardRef(ListItemSecondaryAction))>
          <WithStyles(Money)
            value={1}
          />
        </WithStyles(ForwardRef(ListItemSecondaryAction))>
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItemText))>
          Sum
        </WithStyles(ForwardRef(ListItemText))>
        <WithStyles(ForwardRef(ListItemSecondaryAction))>
          <WithStyles(Money)
            value={101}
          />
        </WithStyles(ForwardRef(ListItemSecondaryAction))>
      </WithStyles(ForwardRef(ListItem))>
    </WithStyles(ForwardRef(List))>
  </div>
</Fragment>
`;

exports[`matches snapshot when loading 1`] = `
<Fragment>
  <div
    className="root"
  >
    <Progress />
  </div>
</Fragment>
`;
